02_bubble/ailyntang/sort_test.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSortingEdgeCases(t *testing.T) {
+	assert := assert.New(t)
+
+	assert.Equal([]int{}, bubble([]int{}), "empty array as input")
+	assert.Equal([]int{4}, bubble([]int{4}), "one integer as input")
+	assert.Equal([]int{0}, bubble([]int{0}), "0 as input")
+}
+
+func TestSorting(t *testing.T) {
+	assert := assert.New(t)
+
+	assert.Equal([]int{1, 11, 111, 222, 333}, bubble([]int{1, 11, 111, 222, 333}), "input is already sorted")
+	assert.Equal([]int{11, 22, 33, 44}, bubble([]int{44, 33, 22, 11}), "input is reverse sorted")
+	assert.Equal([]int{1, 11, 22, 55}, bubble([]int{11, 55, 22, 1}), "positive numbers")
+	assert.Equal([]int{-33, -22, -1}, bubble([]int{-22, -33, -1}), "negative numbers")
+	assert.Equal([]int{-88, -3, 0, 3, 88}, bubble([]int{88, -3, -88, 0, 3}), "positive and negative numbers")
+	assert.Equal([]int{-5, -5, -5, -2, -2, 3}, bubble([]int{-2, 3, -2, -5, -5, -5}), "duplicate numbers in input")
```

Comment:
I would suggest including a test case with a large input array. I did this with your solution and found that it doesn't actually sort correctly. If you test with an input array of size ten at least, you should see some failures

Reply:
Great pick up, thanks for that. I've added an array with 19 elements to the test case. I'll keep this in mind for future labs and tests.